pkg/object/graph/depends.go

```diff
@@ -83,20 +83,27 @@ func ReverseSortObjs(objs object.UnstructuredSet) ([]object.UnstructuredSet, err
 // addApplyTimeMutationEdges updates the graph with edges from objects
 // with an explicit "apply-time-mutation" annotation.
 func addApplyTimeMutationEdges(g *Graph, objs object.UnstructuredSet) {
-	for _, obj := range objs {
-		id := object.UnstructuredToObjMetaOrDie(obj)
+	ids := object.ObjMetadataSet(object.UnstructuredsToObjMetasOrDie(objs))
+	for i, obj := range objs {
+		id := ids[i]
 		klog.V(3).Infof("adding vertex: %s", id)
 		g.AddVertex(id)
 		if mutation.HasAnnotation(obj) {
 			subs, err := mutation.ReadAnnotation(obj)
 			if err != nil {
-				// TODO: fail task if parse errors?
-				klog.V(3).Infof("failed to add edges from: %s: %s", id, err)
+				klog.Warningf("warning: failed to sort explicit dependency: %v",
```

Comment:
`klog.Warning` prints directly to the user when using kapply or kpt, but for server-side usage it would just print to the service log. 

Should we make a new WarningEvent instead, so the printer can handle it?

Reply:
The `Warning` event would be a new top-level event type then? It seems like it would be only for the user to understand what is going on, as it seems difficult to use it for anything useful from code? Another option (perhaps in addition to a `Warning` event) would be to allow users to specify if invalid references should lead to an error or not. Although that would require that we can fit the behavior into the continue-on-error strategy, but this happens before we apply any of the resources.

Reply:
So I think the question here is what should happen if either the annotation can't be read or it doesn't specify a namespace in a reference to a namespace-scoped resource. If we print an error or send an error event, then presumably we will go on with applying the resource right? I think considering the resource as failed/skipped in this situation might be more appropriate, and we can provide information back to the caller through the `Error` property on the `ApplyEvent`. It seems that applying the resource without doing the apply-time mutation almost certainly will result in a different outcome than the user expected. And by just considering that particular resource as failed/skipped, we will still continue applying other resources.

Reply:
I've defined the problem and a couple proposals here: https://github.com/kubernetes-sigs/cli-utils/issues/484